### PR TITLE
Reasoner cleanup

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

**Comment by kasper-piskorski** [comment]
please retest this please

**Comment by fppt** [comment]
@kasper-piskorski Rebase please so we can get this one in.

**Comment by kasper-piskorski** [comment]
failing engine test: io.mindmaps.engine.controller.ImportControllerTest.testLoadOntologyAndDataDistributed

**Comment by kasper-piskorski** [comment]
please retest this please

File: mindmaps-graql/src/main/java/io/mindmaps/graql/Reasoner.java [hunk 1]
```diff
@@ -204,7 +130,7 @@ private QueryAnswers answerWM(AtomicQuery atomicQuery, Set<AtomicQuery> subGoals
 
         if(queryAdmissible) {
             Atom atom = atomicQuery.getAtom();
-            Set<Rule> rules = getApplicableRules(atom);
+            Set<Rule> rules = atom.getApplicableRules();
```
Comment (fppt) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/Reasoner.java]: Yay for object orientated programming ! :)

**Pull request merged**